packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -6,6 +6,55 @@ import 'package:meta/meta.dart';
 
 import 'print.dart';
 
+/// The various priority levels used to filter which diagnostics are shown and
+/// omitted.
+///
+/// Trees of Flutter diagnostics can be very large so filtering the diagnostics
+/// shown matters. Typically filtering to only show diagnostics with at least
+/// level [debug] is appropriate.
+enum DiagnosticLevel {
+  /// Diagnostics that should not be shown.
+  ///
+  /// For example, property values that convey information which is completely
+  /// redundant with other properties or properties where the value is
+  /// misleading or hard to understand.
+  hidden,
+
+  /// Trivial or redundant diagnostics information.
+  ///
+  /// Use this level for diagnostic properties that match their default value
+  /// and other cases where showing a diagnostic would not add value.
+  fine,
+
+  /// Diagnostics that should only be shown when performing fine grained
```

Comment:
the use of the word "fine" in the definition of "debug" is a bit suggestive too.

Reply:
sorry about the bad comments. Updated them to convey the actual distinction I'm getting at. Generally, if you set the minLevel to fine, you should still see beautiful diagnostics just a lot more info and some diagnostics that maybe if you were an experienced user you wouldn't be interested in as they are the default value.
If you set minLevel to hidden you see a bunch more kinda ugly looking and barely functional diagnostics, but you are displaying diagnostics that the component authors explicitly decided to hide so it is your own fault.
If you set minLevel to debug you have everything you need to debug assuming you are savvy enough to understand what the defaultValues are, etc.
If you set minLevel to info you should have enough info to browse the structure.